lib/collection/src/operations/verification/mod.rs

```diff
@@ -251,3 +251,184 @@ impl StrictModeVerification for SearchParams {
         None
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::sync::Arc;
+
+    use common::cpu::CpuBudget;
+    use segment::types::{
+        Condition, FieldCondition, Filter, Match, SearchParams, StrictModeConfig,
+    };
+    use tempfile::Builder;
+
+    use super::StrictModeVerification;
+    use crate::collection::{Collection, RequestShardTransfer};
+    use crate::config::{CollectionConfig, CollectionParams, WalConfig};
+    use crate::operations::point_ops::{FilterSelector, PointsSelector};
+    use crate::operations::shared_storage_config::SharedStorageConfig;
+    use crate::operations::types::{
+        CollectionError, CountRequestInternal, DiscoverRequestInternal,
+    };
+    use crate::optimizers_builder::OptimizersConfig;
+    use crate::shards::channel_service::ChannelService;
+    use crate::shards::collection_shard_distribution::CollectionShardDistribution;
+    use crate::shards::replica_set::{AbortShardTransfer, ChangePeerState};
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_strict_mode_verification_trait() {
+        let collection = fixture().await;
+
+        test_query_limit(&collection).await;
+        test_search_params(&collection).await;
+        test_filter_read(&collection).await;
+        test_filter_write(&collection).await;
+        test_request_exact(&collection).await;
+    }
+
+    async fn test_query_limit(collection: &Collection) {
+        assert_strict_mode_error(discovery_fixture(Some(10), None, None), collection).await;
```

Comment:
Shall we also assert a good example here, rather than just a bad example? (also for the others)